oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/DocumentPropertyState.java

```diff
@@ -38,24 +44,66 @@
 import org.apache.jackrabbit.oak.plugins.memory.StringPropertyState;
 import org.apache.jackrabbit.oak.plugins.value.Conversions;
 import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * PropertyState implementation with lazy parsing of the JSOP encoded value.
  */
 final class DocumentPropertyState implements PropertyState {
 
+    private static final Logger LOG = LoggerFactory.getLogger(DocumentPropertyState.class);
+
     private final DocumentNodeStore store;
 
     private final String name;
 
     private final String value;
 
     private PropertyState parsed;
+    private final byte[] compressedValue;
+    private final Compression compression;
+
+    private static final int DEFAULT_COMPRESSION_THRESHOLD = Integer.getInteger("oak.mongo.compressionThreshold", -1);
 
     DocumentPropertyState(DocumentNodeStore store, String name, String value) {
-        this.store = store;
-        this.name = name;
-        this.value = value;
+        this(store, name, value, Compression.GZIP);
+    }
+
+    DocumentPropertyState(DocumentNodeStore store, String name, String value, Compression compression) {
+        if (DEFAULT_COMPRESSION_THRESHOLD == -1) {
+            this.store = store;
+            this.name = name;
+            this.value = value;
+            this.compression = null;
+            this.compressedValue = null;
+        } else {
+            this.store = store;
+            this.name = name;
+            this.compression = compression;
+
+            int size = value.getBytes().length;
```

Comment:
That serializes the String into a byte array, obtains the lenght, and then throws the byte array away.

Reply:
(also it needs to specify the charset)

Reply:
Done